hooks/tk-multi-publish2/maya/publish_files_integ.py

```diff
@@ -0,0 +1,370 @@
+        # check for any group nodes apart from the once mentioned
+        if extras:
+            self.logger.error("Incorrect naming or extra group nodes present in the scene.",
+                              extra={
+                                  "action_show_more_info": {
+                                      "label": "Show Info",
+                                      "tooltip": "Show the conflicting group nodes",
+                                      "text": "Please check the group nodes:\n{}".format("\n".join(extras)) +
+                                              "\n\nOnly following group nodes should be present:\n{}".format(
+                                                  "\n".join(GROUP_NODES))
+                                  }
+                              }
+                              )
+            return False
+        # check if any of the required group nodes are missing
+        elif not set(GROUP_NODES).issubset(set(groups)):
+            self.logger.error("Please ensure all the group nodes are present.",
+                              extra={
+                                  "action_show_more_info": {
+                                      "label": "Show Info",
+                                      "tooltip": "Group nodes",
+                                      "text": "Following group nodes should be present:\n{}".format(
+                                          "\n".join(GROUP_NODES))
+                                  }
+                              }
+                              )
+            return False
+        return True
+
+
+    @staticmethod
+    def _is_group(node=None):
+        """Check for all the group nodes present in the scene.
+            :param:
+                node: all the nodes in the scene
+            :return: True if yes false otherwise
+        """
+        if cmds.nodeType(node) != "transform":
+            return False
+
+        children = cmds.listRelatives(node, c=True)
+        if not children:
+            return True
+
+        for c in children:
+            if cmds.nodeType(c) != 'transform':
+                return False
+        else:
+            return True
+
+
+    def validate(self, task_settings, item):
+        """
+        Validates the given item to check that it is ok to publish. Returns a
+        boolean to indicate validity.
+
+        :param task_settings: Dictionary of Settings. The keys are strings, matching
+            the keys returned in the settings property. The values are `Setting`
+            instances.
+        :param item: Item to process
+        :returns: True if item is valid, False otherwise.
+        """
+        all_dag_nodes = cmds.ls(dag=True, sn=True)
+        groups = [g for g in all_dag_nodes if self._is_group(g)]
+
+        status = True
+        # Checks for the scene file, i.e if the item is not a sequence or a cache file
+        if item.properties['work_path_template'] == 'tk-maya_shot_work_file':
```

Comment:
This is a bad hard-coding. Use item.type instead to filter just the scene file item.